packages/twenty-server/src/engine/metadata-modules/flat-field-metadata/validators/utils/validate-morph-relation-creation-payload.util.ts

```diff
@@ -0,0 +1,122 @@
+import { type RelationCreationPayload } from 'twenty-shared/types';
+import { t } from '@lingui/core/macro';
+import { isDefined } from 'class-validator';
+
+import { FieldMetadataExceptionCode } from 'src/engine/metadata-modules/field-metadata/field-metadata.exception';
+import {
+  type FailedFieldInputTranspilation,
+  type SuccessfulFieldInputTranspilation,
+  type FieldInputTranspilationResult,
+} from 'src/engine/metadata-modules/flat-field-metadata/types/field-input-transpilation-result.type';
+import { validateRelationCreationPayload } from 'src/engine/metadata-modules/flat-field-metadata/validators/utils/validate-relation-creation-payload.util';
+import { type FlatObjectMetadataMaps } from 'src/engine/metadata-modules/flat-object-metadata-maps/types/flat-object-metadata-maps.type';
+import { type FlatObjectMetadata } from 'src/engine/metadata-modules/flat-object-metadata/types/flat-object-metadata.type';
+
+type RelationCreationPayloadAndObjectMetadata = {
+  relationCreationPayload: RelationCreationPayload;
+  targetFlatObjectMetadata: FlatObjectMetadata;
+};
+
+type ValidateMorphRelationCreationPayloadUtilArgs = {
+  morphRelationCreationPayload: RelationCreationPayload[];
+  existingFlatObjectMetadataMaps: FlatObjectMetadataMaps;
+};
+export const validateMorphRelationCreationPayload = async ({
+  existingFlatObjectMetadataMaps,
+  morphRelationCreationPayload,
+}: ValidateMorphRelationCreationPayloadUtilArgs): Promise<
+  FieldInputTranspilationResult<RelationCreationPayloadAndObjectMetadata[]>
+> => {
+  const allRelationType = [
+    ...new Set(
+      morphRelationCreationPayload.map(
+        (relationCreationPayload) => relationCreationPayload.type,
+      ),
+    ),
+  ];
+
+  if (allRelationType.length > 1) {
+    return {
+      status: 'fail',
+      error: {
+        code: FieldMetadataExceptionCode.FIELD_METADATA_RELATION_MALFORMED,
+        message: 'Morph relation relations must have the same relation type',
+        userFriendlyMessage: t`Morph relation relations must have the same relation type`,
+      },
+    };
+  }
+
+  const allRelatedObjectMetadataIds = morphRelationCreationPayload.map(
+    (relationCreationPayload) => relationCreationPayload.targetObjectMetadataId,
+  );
+  const allRelatedObjectMetadataIdsSet = [
+    ...new Set(allRelatedObjectMetadataIds),
+  ];
+
+  if (
+    allRelatedObjectMetadataIds.length !== allRelatedObjectMetadataIdsSet.length
```

Comment:
we could also make sure that all sourceObjectMetadataIds are equal, with something like that

const allSourceObjectMetadataIds = morphRelationCreationPayload.map(
    (relationCreationPayload) => relationCreationPayload.sourceObjectMetadataId,
  );
  const allSourceObjectMetadataIdsSet = [
    ...new Set(allSourceObjectMetadataIds),
  ];

  if (
    allSourceObjectMetadataIdsSet.length !== 1
  )

Reply:
That's not what we expect, we especially don't want them to be equal